OJS.Workers.ExecutionStrategies/CSharpProjectTestsExecutionStrategy.cs

```diff
@@ -202,14 +199,14 @@ protected void SaveTestFiles(IEnumerable<TestContext> tests, string compileDirec
 
             var errorsByFiles = this.GetTestErrors(processExecutionResult.ReceivedOutput);
 
-            if (failedTestsCount != errorsByFiles.Count || totalTestsCount != executionContext.Tests.Count())
+            if (failedTestsCount != errorsByFiles.Count || totalTestsCount != executionContext.Input.Tests.Count())
```

Comment:
executionContext.Tests is used multiple times. It is better to use `.ToList()` a single time, because now iteration is performed multiple times, i.e. performance is lower